client/components/domains/use-your-domain-step/index.tsx

```diff
@@ -28,442 +26,323 @@ import {
-
-		return translate( 'Sale price is %(cost)s', { args: { cost: domainProductSalePrice } } );
-	};
-
-	getTransferPriceText = () => {
-		const {
-			cart,
-			currencyCode,
-			translate,
-			domainsWithPlansOnly,
-			isSignupStep,
-			productsList,
-			selectedSite,
-		} = this.props;
-		const { searchQuery } = this.state;
-		const productSlug = getDomainProductSlug( searchQuery );
-		const domainsWithPlansOnlyButNoPlan =
-			domainsWithPlansOnly && ( ( selectedSite && ! isPlan( selectedSite.plan ) ) || isSignupStep );
-
-		const domainProductPrice = getDomainPrice( productSlug, productsList, currencyCode );
-
-		if (
-			domainProductPrice &&
-			( isNextDomainFree( cart ) ||
-				isDomainBundledWithPlan( cart, searchQuery ) ||
-				domainsWithPlansOnlyButNoPlan ||
-				getDomainTransferSalePrice( productSlug, productsList, currencyCode ) )
+			domainsWithPlansOnlyButNoPlan ||
+			getDomainTransferSalePrice( productSlug, productsList, currencyCode )
 		) {
-			return translate( 'Renews at %(cost)s', { args: { cost: domainProductPrice } } );
-		}
-
-		if ( domainProductPrice ) {
-			return translate( '%(cost)s per year', { args: { cost: domainProductPrice } } );
-		}
-	};
-
-	getMappingPriceText = () => {
-		const {
-			cart,
-			currencyCode,
-			domainsWithPlansOnly,
-			primaryWithPlansOnly,
-			productsList,
-			selectedSite,
-			translate,
-		} = this.props;
-		const { searchQuery } = this.state;
-
-		let mappingProductPrice;
-
-		const price = get( productsList, [ 'domain_map', 'cost' ], null );
-		if ( price ) {
-			mappingProductPrice = formatCurrency( price, currencyCode );
-			mappingProductPrice = translate(
-				'%(cost)s per year plus registration costs at your current provider',
-				{ args: { cost: mappingProductPrice } }
-			);
+			transferPriceText = translate( 'Renews at %(cost)s', { args: { cost: domainProductPrice } } );
+		} else {
+			transferPriceText = translate( '%(cost)s per year', { args: { cost: domainProductPrice } } );
 		}
+	}
 
-		if (
-			isDomainMappingFree( selectedSite ) ||
-			isNextDomainFree( cart ) ||
-			isDomainBundledWithPlan( cart, searchQuery )
-		) {
-			mappingProductPrice = translate(
-				'Free with your plan, but registration costs at your current provider still apply'
-			);
-		} else if ( domainsWithPlansOnly || primaryWithPlansOnly ) {
-			mappingProductPrice = translate(
-				'Included in annual paid plans, but registration costs at your current provider still apply'
-			);
-		}
+	let mappingPriceText;
+	const price = get( productsList, [ 'domain_map', 'cost' ], null );
+	if ( price ) {
+		// @ts-expect-error despite the TS error, formatCurrency works with a
+		// `null` currencyCode and uses a fallback currency.
+		mappingPriceText = formatCurrency( price, currencyCode );
+		mappingPriceText = translate(
+			'%(cost)s per year plus registration costs at your current provider',
+			{ args: { cost: mappingPriceText } }
+		);
+	}
+	if (
+		isDomainMappingFree( selectedSite ?? undefined ) ||
```

Comment:
Quick fix since `selectedSite` can be `undefined` OR `null`, but `isDomainMappingFree` doesn't accept `null`

Reply:
https://github.com/Automattic/wp-calypso/pull/101854/commits/d637096357993e001d4c96b85c118040515ba736